homeassistant/components/alexa/handlers.py

```diff
@@ -764,9 +765,25 @@ async def async_api_stop(
     entity = directive.entity
     data: dict[str, Any] = {ATTR_ENTITY_ID: entity.entity_id}
 
-    await hass.services.async_call(
-        entity.domain, SERVICE_MEDIA_STOP, data, blocking=False, context=context
-    )
+    if entity.domain == cover.DOMAIN:
```

Comment:
I am not really up to date on the alexa codebase, but if you could add some more context for me:

Does this now use the Stop command from the media player to stop a cover? Shouldn't we stick to what Alexa officially supports?

Reply:
Alexa works with controllers and device classes. Some combinations work, others do not. For `valve` an `AlexaToggleController` is used, and there are othe creative uses to allow integration. But this does not seem to work for a `cover`. The `AlexaPlaybackController` seems to work with cover, with only `Stop` enabled.
This allows to ask Alexa to stop the cover.

https://github.com/home-assistant/core/blob/2f1c1d66cb1aa2fdb46eb4788739e7c7ba2e4cbb/homeassistant/components/alexa/entities.py#L1021-L1022

Reply:
I tried the `ToggleController`, but Alexa won't accept the discovery of the cover with it.